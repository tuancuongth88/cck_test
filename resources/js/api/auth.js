import axios from '@/utils/axios';

export const login = async(params = {}) => {
  return await axios.post('auth/login', params);
};
