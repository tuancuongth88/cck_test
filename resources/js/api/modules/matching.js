import axios from '@/utils/axios';

// Search form
export const getOptionHrOrganization = async(params = {}) => {
  return await axios.get('/hr-org', { params });
};

// Entry
export const getListEntry = async(params = {}) => {
  return await axios.get('/entries', { params });
};
export const deleteMultipleEntry = async(params = {}) => {
  return await axios.post('/entry/hide', params);
};
export const getDetailEntry = async(id) => {
  return await axios.get(`/entry/${id}`);
};

export const updateEntry = async(params = {}) => {
  return await axios.post(`/entry/update-status/${params.id}`, params);
};

// offer
export const getListOffer = async(params = {}) => {
  return await axios.get('/offer', { params });
};

export const deleteMultipleOffer = async(params = {}) => {
  return await axios.post('/offer/remove-offer', params);
};

export const getDetailOffer = async(id) => {
  return await axios.get(`/offer/${id}`);
};

export const updateOffer = async(params = {}) => {
  return await axios.post(`/offer/update-status/${params.id}`, params);
};

// interview
export const getListInterview = async(params = {}) => {
  return await axios.get('/interview', { params });
};

export const deleteMultipleInterview = async(params = {}) => {
  return await axios.post('/interview/hide', params);
};

export const putSetupCalender = async(params = {}) => {
  return await axios.put(`/interview/setup-calendar/${params.id}`, params.calender);
};

export const putConfirmCalender = async(params = {}) => {
  return await axios.put(`/interview/confirm-calendar/${params.id}`, params);
};

export const putDeclineInterview = async(params = {}) => {
  return await axios.put(`/interview/confirm-interview-hr-decline/${params.id}`, params);
};

export const putCancelInterview = async(params = {}) => {
  return await axios.put(`/interview/confirm-interview-company-cancel/${params.id}`, params);
};

export const putSetupZoom = async(params = {}) => {
  return await axios.put(`/interview/setup-zoom/${params.id}`, params);
};

export const putConfirmReview = async(params = {}) => {
  return await axios.put(`/interview/confirm-interview-company-review/${params.id}`, params);
};

export const getDetailInterview = async(id) => {
  return await axios.get(`/interview/${id}`);
};

// result
export const getListResult = async(params = {}) => {
  return await axios.get('/result', { params });
};

export const deleteMultipleResult = async(params = {}) => {
  return await axios.post('/result/hide', params);
};

export const getDetailResult = async(id) => {
  return await axios.get(`/result/${id}`);
};

export const updateResult = async(params = {}) => {
  return await axios.put(`/result/${params.id}`, params);
};
