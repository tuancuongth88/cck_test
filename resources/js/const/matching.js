const account_classification_option = [
  { key: 1, value: '申請中', translate: 'Requesting' },
  { key: 2, value: '却下', translate: 'Reject' },
  { key: 3, value: '辞退', translate: 'Decline' },
  { key: 4, value: '他社内定', translate: 'Other company official offer	' },
];

const reasonResultOptions = [
  { key: '他社内定', value: '他社内定' },
  { key: '就業継続（転職断念）', value: '就業継続（転職断念）' },
  { key: '条件不一致（年収）', value: '条件不一致（年収）' },
  { key: '条件不一致（勤務地）', value: '条件不一致（勤務地）' },
  { key: '条件不一致（その他待遇）', value: '条件不一致（その他待遇）' },
  { key: '就業環境', value: '就業環境' },
  { key: '家族の反対', value: '家族の反対' },
  { key: '在留資格不一致', value: '在留資格不一致' },
  { key: '本人との連絡不通 ', value: '本人との連絡不通 ' },
  { key: 'その他', value: 'その他' },
];

const reasonOfferOptions = [
  { key: '他社内定', value: '他社内定' },
  { key: '就業継続（転職断念）', value: '就業継続（転職断念）' },
  { key: '条件不一致（年収）', value: '条件不一致（年収）' },
  { key: '条件不一致（勤務地）', value: '条件不一致（勤務地）' },
  { key: '条件不一致（その他待遇）', value: '条件不一致（その他待遇）' },
  { key: '就業環境', value: '就業環境' },
  { key: '家族の反対', value: '家族の反対' },
  { key: '在留資格不一致', value: '在留資格不一致' },
  { key: '本人との連絡不通 ', value: '本人との連絡不通 ' },
  { key: 'その他', value: 'その他' },
];

const ENTRY_STATUS = {
  REQUESTING: 1,
  REJECT: 2,
  DECLINE: 3,
  OTHER_COMPANY_OFFICAL_OFFER: 4, // figma bỏ cái này
  CONFIRM: 4,
};

const OFFER_STATUS = {
  REQUESTING: 1,
  DECLINE: 2,
  CONFIRM: 3,
};

const RESULT_STATUS = {
  OFFICAL_OFFER: 1,
  NOT_PASS: 2,
  OFFICAL_OFFER_CONFIRM: 3,
  OFFICAL_OFFER_DECLINE: 4,
  EXPIRATION: -1,
};

export {
  account_classification_option,
  reasonResultOptions,
  reasonOfferOptions,
  ENTRY_STATUS,
  OFFER_STATUS,
  RESULT_STATUS
};
