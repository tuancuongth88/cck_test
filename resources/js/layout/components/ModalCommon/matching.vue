<template>
  <b-modal
    id="id-modal-common"
    :ref="refs"
    :size="size"
    hide-footer
    static
    no-fade
    centered
    :modal-key="refs"
    @hidden="closeModal"
  >
    <template #modal-header>
      <button type="button" class="close" @click="closeModal">
        <span aria-hidden="true">&times;</span>
      </button>
    </template>
    <div class="content-modal">
      <slot name="body" />
    </div>
  </b-modal>
</template>

<script>
import EventBus from '@/utils/eventBus';

export default {
  name: 'ModalCommon',
  components: {},

  props: {
    refs: {
      type: String,
      default: '',
      required: true,
    },
    // show: {
    //   type: Boolean,
    //   default: false,
    //   required: true,
    // },
    size: {
      type: String,
      default: '',
      required: true,
    },
  },

  data() {
    return {};
  },

  computed: {},

  created() {
    EventBus.$on('open-modal', (param) => {
      if (Object.keys(this.$refs)[0] === param) {
        if (!this.$refs[param]) {
          return 0;
        } else {
          this.$refs[param].show();
        }
      }
    });

    EventBus.$on('close-modal', (param) => {
      // this.$refs[this.refs].hide();
      // this.$emit('reset-modal');
      if (Object.keys(this.$refs)[0] === param) {
        if (!this.$refs[param]) {
          return 0;
        } else {
          this.$refs[param].hide();
          this.$emit('reset-modal');
        }
      }
    });
  },

  methods: {
    closeModal() {
      this.$refs[this.refs].hide();
      this.$emit('reset-modal');
    },
  },
};
</script>

<style lang="scss" scoped>
@import '@/scss/_variables.scss';
@import '@/scss/modules/common/common.scss';
</style>
