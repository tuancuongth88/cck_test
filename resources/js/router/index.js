import Vue from 'vue';
import VueRouter from 'vue-router';

Vue.use(VueRouter);

import homeManagement from './modules/homeManagement.js';
import onGoingJob from './modules/showMoreOnGoingJob.js';
import hrManagement from './modules/hrManagement.js';
import hrOrganization from './modules/hrOrganization.js';
import companyManagement from './modules/companyManagement.js';
import matchingManagement from './modules/matchingManagement.js';
import jobManagement from './modules/jobManagement.js';
import jobSearch from './modules/jobSearch.js';
import hrSearch from './modules/hrSearch.js';
import favorite from './modules/favorite.js';
import changePassword from './modules/changePassword.js';

export const constantRoutes = [
  {
    path: '/login',
    name: 'Login',
    component: () => import(/* webpackChunkName: "Login" */ '@/pages/Login/index.vue'),
    meta: {
      title: 'ROUTER_LOGIN',
    },
    hidden: true,
  },
  // Link follow gmail from BE
  {
    path: '/api/auth/password-reset',
    name: 'Reset password confirm',
    component: () => import('@/pages/PassWords/ResetPassWord/index.vue'),
    meta: {
      title: 'RESET_PASS_WORD_TITLE',
    },
    hidden: true,
  },
  {
    path: '/forget-password',
    name: 'Reset password',
    component: () => import('@/pages/PassWords/ForgetPassWord/index.vue'),
    meta: {
      title: 'RESET_PASS_WORD_TITLE',
    },
    hidden: true,
  },
  {
    path: '/human-resourse-register',
    name: 'human-resourse-register',
    component: () => import('@/pages/RegisterHrOrigin/index.vue'),
    meta: {
      title: 'RegisterHrOrigin',
    },
    hidden: true,
  },
  {
    path: '/company-register',
    name: 'company-register',
    component: () => import('@/pages/RegisterCompany/index.vue'),
    meta: {
      title: 'Company register',
    },
    hidden: true,
  },
  // {
  //   path: '/not-permission',
  //   component: () => import('@/pages/Error/NotPermission.vue'),
  //   meta: {
  //     title: 'Permission not Found',
  //   },
  //   hidden: true,
  // },
];

export const asyncRoutes = [
  homeManagement,
  onGoingJob,
  hrSearch,
  hrManagement,
  hrOrganization,
  companyManagement,
  jobManagement,
  jobSearch,
  matchingManagement,
  favorite,
  changePassword,
  {
    path: '/',
    redirect: { name: 'Home' },
    hidden: true,
  },
];

const createRouter = () => new VueRouter({
  mode: 'history',
  scrollBehavior: () => ({ y: 0 }),
  routes: constantRoutes,
});

const router = createRouter();

export function resetRouter() {
  const newRouter = createRouter();
  router.matcher = newRouter.matcher;
}

export default router;
