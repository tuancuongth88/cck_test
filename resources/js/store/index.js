import Vue from 'vue';
import Vuex from 'vuex';

import app from './modules/app';
import job from './modules/job';
import user from './modules/user';
import message from './modules/message';
import jobSearch from './modules/jobSearch';
import permission from './modules/permission';
import hrSearchQuery from './modules/hrSearchQuery';
import userManagement from './modules/userManagement';
import routerParam from './modules/routerParam';

import getters from './getters';

Vue.use(Vuex);

const modules = {
  app,
  job,
  user,
  message,
  jobSearch,
  permission,
  hrSearchQuery,
  userManagement,
  routerParam,
};

const store = new Vuex.Store({
  modules,
  getters,
});

export default store;
