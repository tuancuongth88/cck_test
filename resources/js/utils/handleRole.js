import Cookies from 'js-cookie';
import constAuthority from '@/const/authority';

/**
 * Change object roles BE response -> Object { role, permission }
 * @param {Object} LIST_ROLE
 * @returns Object
 */
export function handleRole(LIST_ROLE) {
  const ROLES = [];
  const PERMISSIONS = [];
  for (const ROLE in LIST_ROLE) {
    ROLES.push(ROLE);

    for (const PERMISSION in LIST_ROLE[ROLE]) {
      PERMISSIONS.push(LIST_ROLE[ROLE][PERMISSION]['name']);
    }
  }

  return {
    ROLES,
    PERMISSIONS,
  };
}

/**
 * Function get Roles in Cookies
 * @returns List Roles
 */
export function getRole() {
  const ROLE = Cookies.get('role');

  if (ROLE) {
    return JSON.parse(ROLE);
  }

  return [];
}

/**
 * Function get Permission in Cookies
 * @returns List Permissions
 */
export function getPermission() {
  const PERMISSION = Cookies.get('permission');

  if (PERMISSION) {
    return JSON.parse(PERMISSION);
  }

  return [];
}

/**
 * Function get List Add Routes in Local Storage
 * @returns List Add Routes
 */
export function getAddRoutes() {
  const ADD_ROUTES = localStorage.getItem('addRoutes');

  if (ADD_ROUTES) {
    return JSON.parse(ADD_ROUTES);
  }

  return [];
}

/**
 * Function get list Routes in Local Storage
 * @returns List Routes
 */
export function getRoutes() {
  const ROUTES = localStorage.getItem('routes');

  if (ROUTES) {
    return JSON.parse(ROUTES);
  }

  return [];
}

/**
 * Function get Current Routes in Local Storage
 * @returns Current Routes
 */
export function getCurrentRoutes() {
  const CURRENT_ROUTES = localStorage.getItem('currentRoutes');

  if (CURRENT_ROUTES) {
    return CURRENT_ROUTES;
  }

  return '/vehicle/list';
}

/**
 * Function get index of authority
 * @param {String} authority
 * @returns Index of authority
 */
export function getAuthority(authority) {
  const LIST_AUTHORITY = constAuthority.AUTHORITY;

  return (LIST_AUTHORITY.indexOf(authority) + 1);
}
