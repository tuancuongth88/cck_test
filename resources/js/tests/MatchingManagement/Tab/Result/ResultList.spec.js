/* eslint-disable no-unused-vars */
import { createLocalVue, shallowMount, mount } from '@vue/test-utils';
import store from '@/store';
import router from '@/router';
import Result from '@/pages/MatchingManagement/Tab/Result.vue';
import MatchingManagement from '@/pages/MatchingManagement/index';
import ROLE from '@/const/role.js';

import { getListResult } from '@/api/modules/matching.js';

import { login } from '@/api/auth';
import { handleRole } from '@/utils/handleRole';

describe('Test component MatchingManagement Tab Result', () => {
  const fieldsResult = [
    {
      key: 'selected',
      sortable: false,
      label: '',
      class: 'bg-header',
      thStyle: {
        backgroundColor: '#1D266A',
        width: '50px',
        color: '#fff',
        textAlign: 'center',
      },
      thClass: 'text-center',
      tdClass: 'text-center',
    },
    {
      key: 'id',
      sortable: true,
      label: 'ID',
      class: 'bg-header',
      thStyle: {
        width: '80px',
        backgroundColor: '#1D266A',
        color: '#fff',
        textAlign: 'center',
      },
      thClass: 'text-center',
    },
    {
      key: 'code',
      sortable: true,
      label: 'エントリー ID',
      class: 'bg-header',
      thStyle: {
        width: '114px',
        backgroundColor: '#1D266A',
        color: '#fff',
        textAlign: 'center',
      },
      tdClass: 'text-center',
    },
    {
      key: 'updating_date',
      sortable: true,
      label: '更新日',
      class: 'bg-header',
      thStyle: {
        width: '120px',
        backgroundColor: '#1D266A',
        color: '#fff',
        textAlign: 'center',
      },
      tdClass: 'text-center',
    },
    {
      key: 'full_name',
      sortable: true,
      label: '氏名',
      class: 'bg-header',
      thStyle: {
        backgroundColor: '#1D266A',
        width: '200px',
        color: '#fff',
        textAlign: 'center',
      },
    },
    {
      key: 'job_title',
      sortable: true,
      label: '求人名',
      class: 'bg-header',
      thStyle: {
        width: '200px',
        backgroundColor: '#1D266A',
        color: '#fff',
        textAlign: 'center',
      },
    },
    {
      key: 'status_selection',
      sortable: true,
      label: '選考 ステータス',
      class: 'bg-header',
      thStyle: {
        width: '120px',
        backgroundColor: '#1D266A',
        color: '#fff',
        textAlign: 'center',
      },
      tdClass: 'text-center',
    },
    {
      key: 'time',
      sortable: true,
      label: '期限',
      class: 'bg-header',
      thStyle: {
        width: '120px',
        backgroundColor: '#1D266A',
        color: '#fff',
        textAlign: 'center',
      },
      tdClass: 'text-center',
    },
    {
      key: 'detail',
      label: '詳細',
      class: 'bg-header',
      thStyle: {
        width: '104px',
        backgroundColor: '#1D266A',
        color: '#fff',
        textAlign: 'center',
      },
      tdClass: 'text-center',
    },
  ];

  const dataResult = [
    {
      id: 7,
      code: '1691053951',
      updating_date: '20230803',
      full_name: 'Dinh',
      job_title: 'IT engineer',
      status_selection: 2,
      time: null,
    },
  ];
  const localVue = createLocalVue();

  test('Case 1: Check render Result Template', async() => {
    const localVue = createLocalVue();
    const wrapper = mount(Result, {
      localVue,
      router,
      store,
      stubs: {
        BIcon: true,
      },
    });

    const dataResult = [
      {
        id: 7,
        code: '1691053951',
        updating_date: '20230803',
        full_name: 'Dinh',
        job_title: 'IT engineer',
        status_selection: 2,
        time: null,
      },
    ];
    await wrapper.setProps({ dataResult: dataResult });

    expect(typeof Result.data).toBe('function');
    const tabResultTemplate = wrapper.findComponent({ name: 'Result' });
    expect(tabResultTemplate.exists()).toBe(true);

    wrapper.destroy();
  });

  test('Case 2: Check render component  Matching management tab Result', async() => {
    const wrapper = mount(Result, {
      localVue,
      router,
      store,
      stubs: {
        BIcon: true,
      },
    });
    const dataResult = [
      {
        id: 7,
        code: '1691053951',
        updating_date: '20230803',
        full_name: 'Dinh',
        job_title: 'IT engineer',
        status_selection: 2,
        time: null,
      },
    ];
    await wrapper.setProps({ dataResult: dataResult });

    const table_hr = wrapper.find('#result-table-list');
    // expect(table_hr).toMatchSnapshot();
    const btnCheckbox = table_hr.find('#checkbox-result');
    expect(btnCheckbox.exists()).toBe(true);

    const TABLE_BODY = table_hr.find('tbody');
    expect(TABLE_BODY.exists()).toBe(true);

    const TABLE_BODY_TR = TABLE_BODY.findAll('tr');
    expect(TABLE_BODY_TR.length).toEqual(1); //

    for (let tr = 0; tr < TABLE_BODY_TR.length; tr++) {
      const LIST_TD = TABLE_BODY_TR.at(tr).findAll('td');
      expect(LIST_TD.at(0).find('input[type="checkbox"]').exists()).toBe(true);
      expect(LIST_TD.at(0).find('input[type="checkbox"]').element.checked).toBe(false);
    }
  });

  test('Case 3: Check API tab Result respone code 200', async() => {
    const wrapper = mount(Result, {
      localVue,
      router,
      store,
      stubs: {
        BIcon: true,
      },
    });

    const params = {
      mail_address: '1okuridashi_hanoi@gmail.vn',
      password: '123456789CCk',
    };

    await login(params).then((response) => {
      if (response['data']['code'] === 200) {
        const TOKEN = response['data']['data']['access_token'];
        const USER = response['data']['data']['profile'];

        const { ROLES, PERMISSIONS } = handleRole([]);

        const USER_INFO = {
          token: TOKEN,
          profile: USER,
          roles: ROLES,
          permissions: PERMISSIONS,
        };

        store.dispatch('user/saveLogin', USER_INFO);
        // expect(store.getters.token).toBe(TOKEN);
        //
      }
    });

    const response = await getListResult();
    const { code } = response.data;
    // console.log('case 5: getListResult :', response);
    expect(code).toBe(200);

    wrapper.destroy();
  });

  // test('Case 4: Check button Delete Mutiple and event function when click button Delete Mutiple', async () => {
  //   const handleDeleteAll = jest.fn();
  //   const wrapper = mount(MatchingManagement, {
  //     localVue,
  //     router,
  //     store,
  //     stubs: {
  //       BIcon: true,
  //     },
  //     methods: {
  //       handleDeleteAll,
  //     },
  //   });

  //   await wrapper.setProps({ dataResult: dataResult });
  //   await wrapper.setData({ tabIndex: 3 });

  //   const BTN_ALL_DELETE = wrapper.find('#btn-delete-all');
  //   expect(BTN_ALL_DELETE.exists()).toBeTruthy();

  //   await BTN_ALL_DELETE.trigger('click');
  //   expect(handleDeleteAll).toHaveBeenCalled();
  // });

  test('Case 5: Check component render Table tab Result have data', async() => {
    const localVue = createLocalVue();
    const wrapper = mount(Result, {
      localVue,
      router,
      store,
      stubs: {
        BIcon: true,
      },
    });
    const dataResultTest = [
      {
        id: 7,
        code: '1691053951',
        updating_date: '20230803',
        full_name: 'Dinh',
        job_title: 'IT engineer',
        status_selection_name: '不合格',
        status_selection: 2,
        time: null,
      },
      {
        id: 8,
        code: '1691053991',
        updating_date: '20230703',
        full_name: 'Dinh 1',
        job_title: 'IT engineer 1',
        status_selection_name: '内定',
        status_selection: 1,
        time: null,
      },
    ];
    await wrapper.setProps({ dataResult: dataResultTest });

    const table_hr = wrapper.find('#result-table-list');
    expect(table_hr).toMatchSnapshot(); //

    const btnCheckbox = table_hr.find('#checkbox-result');
    expect(btnCheckbox.exists()).toBe(true);

    for (let field = 0; field < fieldsResult.length; field++) {
      expect(table_hr.props('fields')[field].key).toEqual(fieldsResult[field].key);
      expect(table_hr.props('fields')[field].sortable).toEqual(fieldsResult[field].sortable);
      // expect(table_hr.props('fields')[field].label).toEqual(fieldsResult[field].label);
      expect(table_hr.props('fields')[field].class).toEqual(fieldsResult[field].class);
      expect(table_hr.props('fields')[field].thStyle).toEqual(fieldsResult[field].thStyle);
    }

    for (let item = 0; item < dataResultTest.length; item++) {
      expect(table_hr.props('items')[item].id).toEqual(dataResultTest[item].id);
      expect(table_hr.props('items')[item].code).toEqual(dataResultTest[item].code);
      expect(table_hr.props('items')[item].updating_date).toEqual(dataResultTest[item].updating_date);
      expect(table_hr.props('items')[item].full_name).toEqual(dataResultTest[item].full_name);
      expect(table_hr.props('items')[item].job_title).toEqual(dataResultTest[item].job_title);
      expect(table_hr.props('items')[item].status_selection).toEqual(dataResultTest[item].status_selection);
      expect(table_hr.props('items')[item].time).toEqual(dataResultTest[item].time);
    }

    const TABLE_BODY = table_hr.find('tbody');
    expect(TABLE_BODY.exists()).toBe(true);

    const TABLE_BODY_TR = TABLE_BODY.findAll('tr');
    expect(TABLE_BODY_TR.length).toEqual(2); // 2 item
    // :disabled="![2, 3].includes(row.item.status)"

    for (let tr = 0; tr < TABLE_BODY_TR.length; tr++) { // find col
      const LIST_TD = TABLE_BODY_TR.at(tr).findAll('td');

      expect(LIST_TD.at(0).find('input[type="checkbox"]').exists()).toBe(true);
      expect(LIST_TD.at(0).find('input[type="checkbox"]').element.checked).toBe(false);
      // expect(LIST_TD.at(1).text()).toBe(String(dataResultTest[tr].id));
      expect(LIST_TD.at(2).text()).toEqual(String(dataResultTest[tr].code));
      expect(LIST_TD.at(3).text()).toEqual(String(dataResultTest[tr].updating_date));
      expect(LIST_TD.at(4).text()).toEqual(String(dataResultTest[tr].full_name));
      expect(LIST_TD.at(5).text()).toEqual(String(dataResultTest[tr].job_title));
      expect(LIST_TD.at(6).text()).toEqual(String(dataResultTest[tr].status_selection_name));
      expect(LIST_TD.at(7).text()).toEqual('-');

      // const LIST_BUTTON = LIST_TD.at(8).findAll('button');
      // expect(LIST_BUTTON.at(8).exists()).toBe(true);
      // expect(LIST_BUTTON.at(8).text()).toEqual('DISPLAY');
      // const spyGoToDetail = jest.spyOn(wrapper.vm, 'goToDetailScreen');

      // await LIST_BUTTON.at(8).trigger('click');
      // expect(spyGoToDetail).toHaveBeenCalledWith(dataResultTest[tr].id);
    }

    // expect(table_hr).toMatchSnapshot();

    // nav-item
  });
});
