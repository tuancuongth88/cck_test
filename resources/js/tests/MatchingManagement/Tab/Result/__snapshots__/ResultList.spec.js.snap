// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component MatchingManagement Tab Result Case 5: Check component render Table tab Result have data 1`] = `
<table id="result-table-list" role="table" aria-busy="false" aria-colcount="9" class="table b-table table-hover">
  <!---->
  <!---->
  <thead role="rowgroup" class="">
    <!---->
    <tr role="row" class="">
      <th role="columnheader" scope="col" aria-colindex="1" aria-label="Selected" class="bg-header text-center" style="background-color: rgb(29, 38, 106); width: 50px; color: rgb(255, 255, 255); text-align: center;">
        <div class="custom-control custom-checkbox"><input type="checkbox" class="custom-control-input" value="true" id="__BVID__233"><label class="custom-control-label" for="__BVID__233"></label></div>
      </th>
      <th role="columnheader" scope="col" tabindex="0" aria-colindex="2" aria-sort="none" class="position-relative bg-header text-center" style="width: 80px; background-color: rgb(29, 38, 106); color: rgb(255, 255, 255); text-align: center;">
        <div>HEADER_ID</div><span class="sr-only"> (Click to sort ascending)</span>
      </th>
      <th role="columnheader" scope="col" tabindex="0" aria-colindex="3" aria-sort="none" class="position-relative bg-header" style="width: 114px; background-color: rgb(29, 38, 106); color: rgb(255, 255, 255); text-align: center;"><label class="mb-0">エントリー <br>
          ID</label><span class="sr-only"> (Click to sort ascending)</span></th>
      <th role="columnheader" scope="col" tabindex="0" aria-colindex="4" aria-sort="none" class="position-relative bg-header" style="width: 120px; background-color: rgb(29, 38, 106); color: rgb(255, 255, 255); text-align: center;">
        <div>HEADER_UPDATING_DATE_ENTRY_MATCHING</div><span class="sr-only"> (Click to sort ascending)</span>
      </th>
      <th role="columnheader" scope="col" tabindex="0" aria-colindex="5" aria-sort="none" class="position-relative bg-header" style="background-color: rgb(29, 38, 106); width: 200px; color: rgb(255, 255, 255); text-align: center;">
        <div>HEADER_FULL_NAME_ENTRY_MATCHING</div><span class="sr-only"> (Click to sort ascending)</span>
      </th>
      <th role="columnheader" scope="col" tabindex="0" aria-colindex="6" aria-sort="none" class="position-relative bg-header" style="width: 200px; background-color: rgb(29, 38, 106); color: rgb(255, 255, 255); text-align: center;">
        <div>HEADER_JOB_LIST_ENTRY_MATCHING</div><span class="sr-only"> (Click to sort ascending)</span>
      </th>
      <th role="columnheader" scope="col" tabindex="0" aria-colindex="7" aria-sort="none" class="position-relative bg-header" style="width: 120px; background-color: rgb(29, 38, 106); color: rgb(255, 255, 255); text-align: center;"><label class="mb-0">選考 <br>
          ステータス</label><span class="sr-only"> (Click to sort ascending)</span></th>
      <th role="columnheader" scope="col" tabindex="0" aria-colindex="8" aria-sort="none" class="position-relative bg-header" style="width: 120px; background-color: rgb(29, 38, 106); color: rgb(255, 255, 255); text-align: center;">
        <div>期限</div><span class="sr-only"> (Click to sort ascending)</span>
      </th>
      <th role="columnheader" scope="col" aria-colindex="9" class="bg-header" style="width: 104px; background-color: rgb(29, 38, 106); color: rgb(255, 255, 255); text-align: center;">
        <div>HEADER_DETAIL_ENTRY_MATCHING</div>
      </th>
    </tr>
  </thead>
  <tbody role="rowgroup">
    <!---->
    <tr role="row" class="">
      <td aria-colindex="1" role="cell" class="bg-header text-center">
        <div class="custom-control custom-checkbox"><input id="checkbox-result" type="checkbox" class="custom-control-input" value="true"><label for="checkbox-result" class="custom-control-label"></label></div>
      </td>
      <td aria-colindex="2" role="cell" class="bg-header"><span></span></td>
      <td aria-colindex="3" role="cell" class="bg-header text-center"><span>
            1691053951
          </span></td>
      <td aria-colindex="4" role="cell" class="bg-header text-center">20230803</td>
      <td aria-colindex="5" role="cell" class="bg-header">
        <div class="w-100 justify-space-between flex-column"><a href="/hr/detail/undefined" class="w-100 justify-space-between flex-column text-dark" target="_self">
            <div title="Dinh" class="one-line-paragraph" style="width: calc(200px - 1.5rem);">
              Dinh
            </div>
            <div class="one-line-paragraph" style="width: calc(200px - 1.5rem);">

            </div>
          </a></div>
      </td>
      <td aria-colindex="6" role="cell" class="bg-header">
        <div class="w-100 justify-space-between flex-column"><a href="/job-search/detail/undefined" class="w-100 justify-space-between flex-column text-dark" target="_self">
            <div class="one-line-paragraph" style="width: calc(200px - 1.5rem);">
              IT engineer
            </div>
          </a></div>
      </td>
      <td aria-colindex="7" role="cell" class="bg-header text-center">
        <!----> <span class="btn-status btn-pending">
            不合格
          </span>
        <!---->
        <!---->
      </td>
      <td aria-colindex="8" role="cell" class="bg-header text-center"><span> - </span></td>
      <td aria-colindex="9" role="cell" class="bg-header text-center"><button id="btn-go-detail" dusk="btn-go-detail-7" class="btn-go-detail"><i class="fas fa-eye icon-detail"></i></button></td>
    </tr>
    <tr role="row" class="">
      <td aria-colindex="1" role="cell" class="bg-header text-center">
        <div class="custom-control custom-checkbox"><input id="checkbox-result" type="checkbox" disabled="disabled" class="custom-control-input" value="true"><label for="checkbox-result" class="custom-control-label"></label></div>
      </td>
      <td aria-colindex="2" role="cell" class="bg-header"><span></span></td>
      <td aria-colindex="3" role="cell" class="bg-header text-center"><span>
            1691053991
          </span></td>
      <td aria-colindex="4" role="cell" class="bg-header text-center">20230703</td>
      <td aria-colindex="5" role="cell" class="bg-header">
        <div class="w-100 justify-space-between flex-column"><a href="/hr/detail/undefined" class="w-100 justify-space-between flex-column text-dark" target="_self">
            <div title="Dinh 1" class="one-line-paragraph" style="width: calc(200px - 1.5rem);">
              Dinh 1
            </div>
            <div class="one-line-paragraph" style="width: calc(200px - 1.5rem);">

            </div>
          </a></div>
      </td>
      <td aria-colindex="6" role="cell" class="bg-header">
        <div class="w-100 justify-space-between flex-column"><a href="/job-search/detail/undefined" class="w-100 justify-space-between flex-column text-dark" target="_self">
            <div class="one-line-paragraph" style="width: calc(200px - 1.5rem);">
              IT engineer 1
            </div>
          </a></div>
      </td>
      <td aria-colindex="7" role="cell" class="bg-header text-center"><span class="btn-status btn-pending">
            内定
          </span>
        <!---->
        <!---->
        <!---->
      </td>
      <td aria-colindex="8" role="cell" class="bg-header text-center"><span> - </span></td>
      <td aria-colindex="9" role="cell" class="bg-header text-center"><button id="btn-go-detail" dusk="btn-go-detail-8" class="btn-go-detail"><i class="fas fa-eye icon-detail"></i></button></td>
    </tr>
    <!---->
    <!---->
  </tbody>
  <!---->
</table>
`;
